From 0c7384499a37e657465f9b4f4b4ef868efda14e2 Mon Sep 17 00:00:00 2001
From: Tsuyoshi Ozawa <ozawa@apache.org>
Date: Wed, 18 Feb 2015 17:32:50 +0900
Subject: [PATCH 2354/2748] HADOOP-11599. Client#getTimeout should use
 IPC_CLIENT_PING_DEFAULT when IPC_CLIENT_PING_KEY
 is not configured. Contributed by zhihai xu.

(cherry picked from commit 3f56a4cb0c57583e285e85a4d0c1584c4de9f1f1)
(cherry picked from commit f10335b08d51ec3edff16ab2e2fbea3dccb5e8bb)

Conflicts:
	hadoop-common-project/hadoop-common/CHANGES.txt
	hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/ipc/TestIPC.java

Change-Id: Ic0d67965e80a86adf375e949ee9e99b12bbb5997
---
 .../main/java/org/apache/hadoop/ipc/Client.java    |    3 ++-
 .../test/java/org/apache/hadoop/ipc/TestIPC.java   |    6 ++++++
 2 files changed, 8 insertions(+), 1 deletion(-)

diff --git a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/ipc/Client.java b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/ipc/Client.java
index c1cde1d..f3fcf87 100644
--- a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/ipc/Client.java
+++ b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/ipc/Client.java
@@ -240,7 +240,8 @@ public static final int getPingInterval(Configuration conf) {
    * @return the timeout period in milliseconds. -1 if no timeout value is set
    */
   final public static int getTimeout(Configuration conf) {
-    if (!conf.getBoolean(CommonConfigurationKeys.IPC_CLIENT_PING_KEY, true)) {
+    if (!conf.getBoolean(CommonConfigurationKeys.IPC_CLIENT_PING_KEY,
+        CommonConfigurationKeys.IPC_CLIENT_PING_DEFAULT)) {
       return getPingInterval(conf);
     }
     return -1;
diff --git a/hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/ipc/TestIPC.java b/hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/ipc/TestIPC.java
index e259a02..f959526 100644
--- a/hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/ipc/TestIPC.java
+++ b/hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/ipc/TestIPC.java
@@ -1238,6 +1238,12 @@ public void run() {
     }
   }
 
+  @Test
+  public void testClientGetTimeout() throws IOException {
+    Configuration config = new Configuration();
+    assertEquals(Client.getTimeout(config), -1);
+  }
+
   private void assertRetriesOnSocketTimeouts(Configuration conf,
       int maxTimeoutRetries) throws IOException {
     SocketFactory mockFactory = Mockito.mock(SocketFactory.class);
-- 
1.7.9.5

